lib/spack/spack/report.py

```diff
@@ -247,30 +260,196 @@ class collect_info(object):
     Raises:
         ValueError: when ``format_name`` is not in ``valid_formats``
     """
-    def __init__(self, specs, format_name, filename):
-        self.specs = specs
+    def __init__(self, format_name, install_command):
         self.format_name = format_name
+        # Consider setting these properties in a more CDash specific place.
+        self.install_command = install_command
+        self.hostname = socket.gethostname()
+        self.osname = platform.system()
```

Comment:
I am not sure that this changes much for the first revision, but note that the OS that Spack builds for can be different from the OS that Spack runs on.  e.g., on Cray and BG/Q.  You can get Spack's identifier for the OS from the Spec.  I think CDash adds some nice icons for the specific os name, so it may make sense to either normalize to what CDash expects, or add more icons to CDash, but that can probably happen later.